src/Symfony/Component/HttpKernel/EventListener/SerializeControllerResultListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Attribute\Serialize;
+use Symfony\Component\HttpKernel\Event\ViewEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Serializer\SerializerInterface;
+
+final class SerializeControllerResultListener implements EventSubscriberInterface
+{
+    private const DEFAULT_FORMAT = 'json';
+
+    public function __construct(private readonly ?SerializerInterface $serializer)
+    {
+    }
+
+    public function onView(ViewEvent $event): void
+    {
+        /** @var Serialize[] $attributes */
+        $attributes = $event->controllerArgumentsEvent->getAttributes()[Serialize::class] ?? [];
+        if (!$attributes) {
+            return;
+        }
+
+        if (!$this->serializer) {
+            throw new \LogicException(sprintf('The "symfony/serializer" component is required to use the "%s" attribute. Try running "composer require symfony/serializer".', Serialize::class));
+        }
+
+        $request = $event->getRequest();
+        $controllerResult = $event->getControllerResult();
+        $format = $request->getPreferredFormat(self::DEFAULT_FORMAT);
+        $attribute = $attributes[0];
+
+        $data = $this->serializer->serialize($controllerResult, $format, $attribute->context);
+
+        $headers = $attribute->headers + ['Content-Type' => $request->getMimeType($format)];
+        $response = new Response($data, $attribute->code, $headers);
+
+        $event->setResponse($response);
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            KernelEvents::VIEW => 'onView',
```

Comment:
how this deals with others listener/subscriber on same event?

Reply:
you'd overwrite on high prio i suppose, not sure if this listener needs to be low prio